fplogd/fplogd.cpp

```diff
@@ -464,13 +489,49 @@ class Impl
                     try
                     {
                         if (log_transport_ && protocol_)
+                        {
                             protocol_->write(str->c_str(), str->size() + 1, 200);
+                            mq_.pop();
+                        } 
+                        else
+                        {
+                            THROW(fplog::exceptions::Transport_Missing);
+                        }
                     }
                     catch(fplog::exceptions::Generic_Exception& e)
                     {
-                        printf("%s\n", e.what().c_str());
-                        if (retries <= 0)
-                            exit(1);
+                        if (retries <= 0) {
+                            fplog::Message error_msg = FPL_ERROR(std::string("Error: " + e.what() + " Log message:" + *str).c_str()).set(fplog::Message::Mandatory_Fields::appname, "fplogd").add(fplog::Message::Optional_Fields::sequence, 0);
```

Comment:
to indicate that this was error from fplog itself, there is a special facility code that should be used